Build failure due to MPL_hash

https://github.com/mpi4py/mpi4py-testing/actions/runs/25717437347/job/75510527704

```
make[2]: Entering directory '/home/runner/work/mpi4py-testing/mpi4py-testing/mpi-build/src/mpl'
  CC       src/mpl_rankmap.lo
  CC       src/arch/mpl_arch_x86.lo
  CC       src/atomic/mpl_atomic.lo
  CC       src/bt/mpl_bt.lo
  CC       src/dbg/mpl_dbg.lo
  CC       src/env/mpl_env.lo
  CC       src/mem/mpl_trmem.lo
  CC       src/mem/mpl_trmem_avx.lo
  CC       src/mem/mpl_trmem_avx512f.lo
  CC       src/msg/mpl_msg.lo
  CC       src/sock/mpl_sock.lo
  CC       src/sock/mpl_sockaddr.lo
  CC       src/sock/mpl_host.lo
  CC       src/str/mpl_str.lo
  CC       src/str/mpl_argstr.lo
  CC       src/str/mpl_arg_serial.lo
  CC       src/str/mpl_hash.lo
  CC       src/misc/mpl_misc.lo
  CC       src/misc/mpl_mkstemp.lo
src/str/mpl_hash.c: In function 'MPL_hash_new':
src/str/mpl_hash.c:40:63: warning: character constant too long for its type
   40 |     hv = (struct MPL_hash *) calloc(1, sizeof(struct MPL_hash));
      |                                                               ^
In file included from ./include/mpl.h:18,
                 from src/str/mpl_hash.c:21:
./include/mpl_trmem.h:243:50: error: expected ';' before '::' token
  243 | #define calloc(a,b)       'Error use MPL_calloc' :::
      |                                                  ^~
src/str/mpl_hash.c:40:30: note: in expansion of macro 'calloc'
   40 |     hv = (struct MPL_hash *) calloc(1, sizeof(struct MPL_hash));
      |                              ^~~~~~
src/str/mpl_hash.c: In function 'f_addto_strpool':
src/str/mpl_hash.c:163:78: error: macro "realloc" passed 2 arguments, but takes just 1
  163 |         p_strpool->pc_pool = realloc(p_strpool->pc_pool, p_strpool->pool_size);
      |                                                                              ^
./include/mpl_trmem.h:245: note: macro "realloc" defined here
  245 | #define realloc(a)        'Error use MPL_realloc' :::
      | 
src/str/mpl_hash.c:163:28: warning: assignment to 'unsigned char *' from incompatible pointer type 'void * (*)(void *, size_t)' {aka 'void * (*)(void *, long unsigned int)'} [-Wincompatible-pointer-types]
  163 |         p_strpool->pc_pool = realloc(p_strpool->pc_pool, p_strpool->pool_size);
      |                            ^
make[2]: *** [Makefile:1188: src/str/mpl_hash.lo] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: Leaving directory '/home/runner/work/mpi4py-testing/mpi4py-testing/mpi-build/src/mpl'
make[1]: *** [Makefile:41056: all-recursive] Error 1
make[1]: Leaving directory '/home/runner/work/mpi4py-testing/mpi4py-testing/mpi-build'
make: *** [Makefile:10050: all] Error 2
Error: Process completed with exit code 2.
```

--- a/src/mpi/datatype/datatype_impl.c
+++ b/src/mpi/datatype/datatype_impl.c
@@ -252,7 +252,7 @@ static MPI_Count MPIR_Type_get_element_bytes(MPI_Count * elements_p,
         case MPI_COMBINER_F90_REAL:
         case MPI_COMBINER_F90_COMPLEX:
         case MPI_COMBINER_F90_INTEGER:
-#ifndef BUILD_MPI_ABI
+#ifndef MPICH_BUILD_MPI_ABI
         case MPI_COMBINER_HVECTOR_INTEGER:
         case MPI_COMBINER_HINDEXED_INTEGER:
         case MPI_COMBINER_STRUCT_INTEGER:

--- a/src/mpl/src/str/mpl_hash.c
+++ b/src/mpl/src/str/mpl_hash.c
@@ -37,7 +37,7 @@ struct MPL_hash *MPL_hash_new(void)
 {
     struct MPL_hash *hv;
 
-    hv = (struct MPL_hash *) calloc(1, sizeof(struct MPL_hash));
+    hv = (struct MPL_hash *) MPL_calloc(1, sizeof(struct MPL_hash), MPL_MEM_OTHER);
     return hv;
 }
 
@@ -160,7 +160,7 @@ int f_addto_strpool(struct strpool *p_strpool, const char *s, int n)
             tn_avg_str_size = p_strpool->i_pool / p_strpool->i_str + 1;
         }
         p_strpool->pool_size = tn_avg_str_size * p_strpool->n_str + n;
-        p_strpool->pc_pool = realloc(p_strpool->pc_pool, p_strpool->pool_size);
+        p_strpool->pc_pool = MPL_realloc(p_strpool->pc_pool, p_strpool->pool_size, MPL_MEM_OTHER);
     }
     memcpy(p_strpool->pc_pool + p_strpool->i_pool, s, n);
     p_strpool->i_str++;

